### PR TITLE
Bytesize transcript dsl2 pipeline starter

### PR DESCRIPTION
**Pull request by FranBonath**

### CODE REVIEW

File: markdown/events/2021/bytesize-24-dsl2-pipeline-starter.md [hunk 1]
```diff
@@ -30,3 +30,93 @@ Slides:
...
+
+[18:26](https://youtu.be/Z_uPj7fAes8&t=1106)
+We will be changing the syntax that we're using for DSL2 very soon, hopefully. We're moving to a more Nextflow native syntax. I've provided a brief description of this in the "contributors to modules" talk at that particular time, if you want to skip through to it, so I won't go through this in any detail. The information is there and hopefully it will make the adoption and usage of these modules even more widely accessible, because we'd be using a native Nextflow syntax. That just removes things like the functions file and other things, that have been a bit of an issue in terms of customization. Watch this space. Things are evolving. Everything will be updated and hopefully you'll be able to keep on top. This is something that we'll probably be discussing and trying to iron out at the hackathon.
+
+[19:22](https://youtu.be/Z_uPj7fAes8&t=1162)
```
Comment (laribritto) [inline markdown/events/2021/bytesize-24-dsl2-pipeline-starter.md:112]:
```suggestion
[19:23](https://youtu.be/Z_uPj7fAes8&t=1162)
```

File: markdown/events/2021/bytesize-24-dsl2-pipeline-starter.md [hunk 2]
```diff
@@ -30,3 +30,93 @@ Slides:
...
...ponents and essentially make things a bit more flexible in terms of pipeline development as well. There's loads of resources so as I mentioned we have bytesize talks, that are revolved around DSL2 pipeline development. How we are tackling DSL2 on nf-core, but there are some generic concepts there, that may be useful to you as well. If you like, in the bytesize tests there are also some guidelines and other stuff that you may be able to take away to use for your own DSL2 pipeline development in non-nf-core pipelines too. Last week, Rike, Maxime and I recorded a couple of more up-to-date talks about the pipeline structure that we have for DSL2 in the template that we maintain on nf-core tools. I just briefly went through that again in preparation for the hackathon, so people get an idea as to what these files are doing and why they're there. Another talk which was typically a 45-minute talk. By my standards, that is where I typically get with these talks. That one was more about the process of picking a module and using the various tooling that we've created, to then contribute that back to nf-core modules. There are some generic concepts there that may be useful to you to take away.
```
[the remark this answers is not in the document]
Comment (FranBonath) [reply markdown/events/2021/bytesize-24-dsl2-pipeline-starter.md:44]: Hej, @laribritto, the change you are suggesting here may be closer to the recording, but they were made deliberately to improve readability. Therefore, I hope it is ok with you if I will not include this change. 
I am very happy for your contribution though!

File: markdown/events/2021/bytesize-24-dsl2-pipeline-starter.md [hunk 3]
```diff
@@ -30,3 +30,93 @@ Slides:
...
+
+[2:43](https://youtu.be/Z_uPj7fAes8&t=163)
+The first thing I would do is attempt to watch some of these talks here, and also we've got loads of documentation and guidelines on the website, that'd be worth looking at. Figuring out how to tackle DSL2 in some way philosophically, would be in comparison to the way that you would attempt to figure out how these components are put together. The smallest unit of a DSL2 pipeline will be a module. So attempting to figure out how you need to mentally create and write these modules, to structure an entire pipeline together, is quite useful. We've got these guidelines and adding new modules and some of the concepts that we've used to attempt to standardize these these modules and the syntax available on the website for you to have a look at.
```
[the remark this answers is not in the document]
Comment (FranBonath) [reply markdown/events/2021/bytesize-24-dsl2-pipeline-starter.md:47]: same here

File: markdown/events/2021/bytesize-24-dsl2-pipeline-starter.md [hunk 4]
```diff
@@ -30,3 +30,93 @@ Slides:
...
+
+[4:45](https://youtu.be/Z_uPj7fAes8&t=285)
+Get the tests working again. With nf-core pipelines we insist on having test data sets for continuous integration and for local testing. It just means that whenever you update the pipeline code you can test the pipeline to make sure it's working. After you've merged in the template sync, try and get the test working again, even if it's still in a DSL1 format. Just try and get the test working again and then decide how you want to tackle the implementation, which i'll go through briefly in some of the following slides. With new nf-core pipelines we have loads of guidelines and docs. There's very small differences in the way that the template has changed between DSL1 and DSL2, in order to manage the modules aspect. More change probably has been in the way that we've siloed away some of the boilerplate code into into lib directories, to make it easier to read the code that we've got there, and to update it. More guidelines and documentations on the website. I won't go through that in detail. Have a look.
```
[the remark this answers is not in the document]
Comment (FranBonath) [reply markdown/events/2021/bytesize-24-dsl2-pipeline-starter.md:56]: and here

File: markdown/events/2021/bytesize-24-dsl2-pipeline-starter.md [hunk 5]
```diff
@@ -30,3 +30,93 @@ Slides:
...
+
+[5:46](https://youtu.be/Z_uPj7fAes8&t=346)
+But most importantly, if you want to contribute a pipeline to nf-core or you're thinking of doing it - whether you start off by with a new nf-core pipeline, or you start off thinking "I'm just going to write my own pipeline but maybe I might contribute to nf-core in the future" - please come and approach us first. Because we try not to have redundancy in pipelines and it's always a community decision as to what gets in. It'd be great if you can approach this before you lay down in your code. Things can become a bit awkward, when you've written a pipeline and it's fully functional and it's in all its awesomeness, but it may not necessarily fit with what we have or what we require. In that case we don't want to disappoint you or make things difficult for everyone.
```
[the remark this answers is not in the document]
Comment (FranBonath) [reply markdown/events/2021/bytesize-24-dsl2-pipeline-starter.md:59]: and here

File: markdown/events/2021/bytesize-24-dsl2-pipeline-starter.md [hunk 6]
```diff
@@ -30,3 +30,93 @@ Slides:
...
+
+[11:16](https://youtu.be/Z_uPj7fAes8&t=676)
+Getting started, I would probably start by looking at existing pipelines that have done this. I'm pointing to nf-core pipelines here, because it's what we know, it's what we've done. There's a full list. If you click on that link there, there's a DSL2 tab on the pipeline health page on the website, that allows you to look at other example pipelines, if they're more applicable to you. I think most importantly is setting up a nice test data set. We try and tackle that right at the beginning. It's always good to test your pipeline right from the offset. It also means that other people can collaborate on the pipeline with you and you can identify bugs and issues and pull requests or locally, that you can fix, whilst developing the pipeline together.
```
[the remark this answers is not in the document]
Comment (FranBonath) [reply markdown/events/2021/bytesize-24-dsl2-pipeline-starter.md:83]: and here

**Review by maxulysse — APPROVED** [review]

**Pull request merged**